src/LanguageServer/Impl/Implementation/ProjectFiles.cs

```diff
@@ -29,15 +29,24 @@ internal sealed class ProjectFiles : IDisposable {
         private readonly ConcurrentDictionary<Uri, IProjectEntry> _projectFiles = new ConcurrentDictionary<Uri, IProjectEntry>();
         private bool _disposed;
 
-        public void Clear() => _projectFiles.Clear();
+        public void Clear() {
+            var entries = _projectFiles.ToArray();
+            _projectFiles.Clear();
+            foreach (var f in entries) {
+                f.Value.Dispose();
+            }
+        }
+
         public IProjectEntry GetOrAddEntry(Uri documentUri, IProjectEntry entry) {
             ThrowIfDisposed();
             return _projectFiles.GetOrAdd(documentUri, entry);
         }
 
         public IProjectEntry RemoveEntry(Uri documentUri) {
             ThrowIfDisposed();
-            return _projectFiles.TryRemove(documentUri, out var entry) ? entry : null;
+            _projectFiles.TryRemove(documentUri, out var entry);
+            entry?.Dispose();
```

Comment:
`ProjectFiles` doesn't create entries or manage their lifetime otherwise, so it's up to the caller to dispose it.

Reply:
Except they don't do it. Maybe it should not be disposable and we can just ignore events from it.